woocommerce/woocommerce-android
# [Tablet support] [Bug] [Beta fix 17.7] Incorrect tablet screen type check

Closes: #11056 


### Description

This PR fixes the wrong implementation of screen type (tablet/phone) detection that resulted in displaying tablet layout on phones in some scenarios. 

### Testing instructions

Steps to reproduce the issue on phones on trunk branch:
1. Go to OS settings > Accessibility > Screen size
2. Set the smallest available display size (this makes screen dp size larger)
3. Observe products and orders screens wrongly display two-pane layouts

Verify there are no regressions on tablets.

### Images/gif

Difference on phones:
|Before|After|
|---|---|
|<video src=https://github.com/woocommerce/woocommerce-android/assets/4527432/bd2d4289-158e-41f4-a0d0-08d7479914ea/>|<video src=https://github.com/woocommerce/woocommerce-android/assets/4527432/036cd3b3-441c-4550-bf07-1bdfdf4cb6b6/>|

On tablet devices behavior should remain the same:

https://github.com/woocommerce/woocommerce-android/assets/4527432/fad89a19-27f4-42a0-a98a-ee9b95283c98
